### ISSUE TITLE
multipart/formData file has 2 extra bytes

### ISSUE THREAD

**Issue by pyramids33** [oakserver/oak#581]
When i upload a file to oak, the file received in the server has 2 extra bytes, ([13,10] which looks like a /r/n.
I confirmed by hashing the files, needed to cut off the last 2 bytes to get matching hashes.

```ts
    import { Application, Context, Router } from "https://deno.land/x/oak@v11.1.0/mod.ts";
    const abortController = new AbortController();
    const app = new Application();
    
    const router = new Router();
    
    router.post('/upload', async (ctx:Context) => {
        const body = ctx.request.body({ type: 'form-data'});
        const form = await body.value.read({ outPath: '.' });
    
        const reqFile = form.files?.[0];
        const size = Deno.statSync(reqFile.filename)?.size;
        console.log('server received', reqFile)
        console.log('server size', size);
        
        const buf = Deno.readFileSync(reqFile.filename);
        console.log('last 2 bytes', buf.slice(-2))
    });
    
    app.use(router.routes())
    
    const serverClosed = app.listen({ 
        hostname: '127.0.0.1', 
        port: 25000,
        signal: abortController.signal
    });
    
    const formData = new FormData();
    const blob = new File([new Uint8Array([ 10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10 ])], 'test.txt');
    console.log('client size', blob.size)     
    
    formData.append('name', 'bgbgks');
    formData.append('677', 'zful ');
    formData.append('filedata', blob);
        
    const res = await fetch('http://127.0.0.1:25000/upload', { 
        signal: abortController.signal, 
        method: 'POST', 
        body: formData
    });
    
    await res.text();
    abortController.abort();
    await serverClosed;
```

**Comment by bidyut18** [comment]
###  Hello @pyramids33, I did some debugging on vs code and found that the data change happened while saving the file form data. It's found during reading.

![Screenshot_2023-04-29-19-56-44_1366x768](https://user-images.githubusercontent.com/69600275/235308635-a1f0f030-9b01-4885-b011-0947e08915ed.png)
 
<p>That 0D is 13 It's added during saving the file. </p>

**Comment by NeKzor** [comment]
Alright, I spent hours trying to figure out where this bug comes from. I tried other frameworks and I even  tried node at some point because I initially thought this was a bug in deno but everything leads to this code piece:

https://github.com/oakserver/oak/blob/7aa02094f86d77d2482225684f4e5a059b93f6dd/multipart.ts#L244-L256

Looks like there was an attempt to fix issue #335 with the PR #338 by @zlumyo but none of the code above makes sense to me:
- **Why is the file truncated based on the newline of the last boundary?**
- **Why does it only truncate the file?** What about the in-memory buffer if you choose to set `maxSize`?

To be fair all of the multipart code is  incorrect when trying to handle new lines. Right now it tolerates `LF` between the data and the boundary part. However, chapter [7.2  The Multipart Content-Type] in RFC1341 clearly states that `CRLF` is required.

[7.2  The Multipart Content-Type]: https://www.w3.org/Protocols/rfc1341/7_2_Multipart.html

My current workaround is to force clients to send extra non-file data after the file part. It's not a great fix but fixing the root cause requires doing it the correct way by actually handling `CRLF` which is not that trivial based on the current multipart logic. Good luck to whoever attempts to go for it.
